packages/core/src/components/axes/ruler.ts

```diff
@@ -50,12 +50,11 @@ export class Ruler extends Component {
 			"[role=graphics-symbol]"
 		);
 		const displayData = this.model.getDisplayData();
-		const domainScale = this.services.cartesianScales.getDomainScale();
 		const rangeScale = this.services.cartesianScales.getRangeScale();
 		const [yScaleEnd, yScaleStart] = rangeScale.range();
 
-		const scaledData: {domainValue: number, originalData: any}[] = displayData.map((d, i) => ({
-			domainValue: this.services.cartesianScales.getDomainValue(d, i),
+		const scaledData: {domainValue: number, originalData: any}[] = displayData.map(d => ({
```

Comment:
why do we need `scaledData`?
The entire logic seems a bit extensive to me.

Maybe I'm missing something but I think we should just use:
```js
const domainIdentifier = this.services.cartesianScales.getDomainIdentifier();
```

and go through `displayData` filtering out datapoints where `d[domainIdentifier]` matches our domain value that the ruler was hovering through

Reply:
we need to call `this.services.cartesianScales.getDomainValue()` on data to compare it with current mouse x position and filter data accordingly, `scaledData` is used basically to avoid having to call getDomainValue multiple times in data and elements manipulations/filtering.
By the way I just pushed a refactor for the reducer, adding an initial filter to only go through points which are within threshold, and also optimized the nearest point matching calcs.

Reply:
Just an additional note: we need to reason in terms of pixels and not in terms of "domain units" because 1 pixel is the maximum resolution we can have. (i.e. two points on the same x coordinate, even if they are shifted by 0.001px, should be matched together) 
If we reason in terms of domain units, we would lose that possibility. 

And being the threshold calculated in px, we couldn't apply that either.

Reply:
ok, the refactor looks ok for now, let's merge than then I'll take another stab at it later.

I just found a few issues though